### PR TITLE
BF: Extract the data rank from the info structure when estimating noise covariance

### PR DESCRIPTION
**Pull request by hoechenberger**
<strike>- Shuffle the data before the cross-validation split. This is required to get any meaningful CV results.</strike>

- Extract the data rank from the `info` structure. At least on my  test data this was necessary to get correct results.

<strike>- Logging and documentation improvements</strike>

### CODE REVIEW

File: 12-make_cov.py [hunk 1]
```diff
@@ -57,8 +58,21 @@ def run_covariance(subject, session=None):
...
+               f'the user-defined value: {config.random_state}')
+    else:
+        msg = 'Using random number generator with default settings'
+
+    print(msg)
+    del msg
+
+    cv = KFold(3, random_state=config.random_state, shuffle=True)
+    cov = mne.compute_covariance(epochs, tmax=0, method='shrunk', cv=cv,
+                                 rank='info')
```
Comment (agramfort) [inline 12-make_cov.py:75]: rank='info' is bug fix but the rest is just doc right?

Comment (hoechenberger) [reply 12-make_cov.py:75]: and `shuffle=True` in the `KFold` instantiation.

FYI this PR contains two commits, one for the bugfix and one for the docs & logging:
bugfix: 2069391
doc: e928ea9

Comment (agramfort) [reply 12-make_cov.py:75]: arff I the present case (with temporally correlated data) I wouldn't not shuffle. You don't want in the test set samples that are "just" like train because of autocorrelated noise. Please keep shuffle=False and make a note. This technique is also known as "block cross-validation" in stats.

Comment (hoechenberger) [reply 12-make_cov.py:65]: Wow I didn't know that! Thanks for the clarification, I'm glad you caught this mistake!

I've reverted those commits (including the documentation & logging updates, which are unnecessary now). I've added `shuffle=False` (which is the default in `sklearn`, but I think being explicit here is a good thing), and added a comment trying to explain _why_ we don't want to shuffle the data.

 Can you please review again?

**Comment by agramfort** [comment]
thx @hoechenberger !
